MaterialDesignThemes.Wpf/Behaviors/PasswordBoxRevealTextBoxBehavior.cs

```diff
@@ -0,0 +1,92 @@
+﻿using System.Reflection;
+using System.Windows.Documents;
+using Microsoft.Xaml.Behaviors;
+
+namespace MaterialDesignThemes.Wpf.Behaviors;
+
+internal class PasswordBoxRevealTextBoxBehavior : Behavior<TextBox>
+{
+    private static readonly DependencyProperty SelectionProperty = DependencyProperty.RegisterAttached(
+        "Selection", typeof(TextSelection), typeof(PasswordBoxRevealTextBoxBehavior), new UIPropertyMetadata(default(TextSelection)));
+    private static void SetSelection(DependencyObject obj, TextSelection? value) => obj.SetValue(SelectionProperty, value);
+    private static TextSelection? GetSelection(DependencyObject obj) => (TextSelection?)obj.GetValue(SelectionProperty);
+
+    internal static readonly DependencyProperty PasswordBoxProperty = DependencyProperty.Register(
+        nameof(PasswordBox), typeof(PasswordBox), typeof(PasswordBoxRevealTextBoxBehavior), new PropertyMetadata(default(PasswordBox)));
+
+    internal PasswordBox? PasswordBox
+    {
+        get => (PasswordBox) GetValue(PasswordBoxProperty);
+        set => SetValue(PasswordBoxProperty, value);
+    }
+
+    protected override void OnAttached()
+    {
+        base.OnAttached();
+        AssociatedObject.IsVisibleChanged += AssociatedObjectOnIsVisibleChanged;
+        if (PasswordBox != null)
+        {
+            var info = typeof(PasswordBox).GetProperty("Selection", BindingFlags.NonPublic | BindingFlags.Instance);
```

Comment:
I would pull this out into a private static property to avoid duplicate reflection calls.
Something like this: 
```C#
private static PropertyInfo SelectionProperty { get; } = typeof(PasswordBox).GetProperty("Selection", BindingFlags.NonPublic | BindingFlags.Instance) ?? throw new InvalidOprationException("Did not find 'Selection' property on PasswordBox");```

Reply:
Sounds reasonable. Will change.